### PR TITLE
Created Book model, Book service and repository

### PR DESCRIPTION
**Pull request by GlebPashko**

### CODE REVIEW

File: src/main/java/org/example/bookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java:11]: Add Table annotation
Don't use Data annotation to entity

File: src/main/java/org/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java:16]: Add required constraints

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:12]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:19]:
```suggestion
```

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert a book: " + book, e);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:35]:
```suggestion
            throw new DataProcessingException("Can't insert a book: " + book, e);
```

File: src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book", Book.class).list();
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java:47]: add catch block

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package org.example.bookstore.service.impl;
+
+import java.util.List;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.BookRepository;
+import org.example.bookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:10]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:17]:
```suggestion
```

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -1 +1,8 @@
-spring.application.name=book-store-git
+spring.application.name=book-store
+spring.datasource.url=jdbc:mysql://localhost:3306/booksdb?serverTimeZone=UTC
+spring.datasource.username=root
+spring.datasource.password=
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties:4]: try to remove if you didn't set up password

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
See comments

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**